retag-for-jenkins

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+set -euo pipefail
+
+if [ $# -lt 1 ] || [ "$1" = "-h" ] || [ "$1" = "--help" ]; then
+  echo "Usage: $0 TAG"
+  echo "Rebases the free-standing git tag <TAG> on the current branch"
+  echo "(make sure you don't have any uncommited local changes)"
+  echo "E.g., after a Jenkins build 'alpha-9999.99.99-mytest' was started based on"
+  echo "the scripts branch 'mybranch' and you have new changes on the branch,"
+  echo "check out your branch and run"
+  echo "  $0 alpha-9999.99.99-mytest"
+  echo
+  echo "TODO: Add feature to update coreos-overlay/portage-stable submodule refs"
+  exit 1
+fi
+
+TAG="$1"
+BRANCH=$(git rev-parse --abbrev-ref HEAD)
+git fetch --tags origin
+BUILD_PATCH=$(git format-patch --output=/dev/stdout "${TAG}~1..${TAG}")
+git checkout --recurse-submodules "${TAG}"
```

Comment:
I'm wondering if it wouldn't be better to do it in a temporary worktree to avoid messing with the current worktree. Like `git workplace add -b tmp-branch /tmp/foo "${TAG}"; cd /tmp/foo; …`?

Reply:
I've never used `git workplace`, will have to look into it - but if it's not a well-known thing I would rather keep the helper at a level where most people can understand what's going on and intervene manually if needed.

Reply:
Sure, fair enough.